test/io.test.js

```diff
@@ -170,18 +188,24 @@ describe('IO', function(){
         describe('readLong()', function(){
             it('should decode and return a long', function(){
                 block.write(new Buffer([0x94, 0x02]));
-                decoder.readLong().should.equal(138);
+                decoder.readLong().equals(138).should.be.true;
```

Comment:
Is a `Long` object coerced into a `Number`? It seems that the [Long](https://github.com/dcodeIO/Long.js/blob/master/dist/Long.js) library doesn't have `valueOf` implemented. Maybe it should:
```javascript
Long.prototype.valueOf = function() {
  return this.toNumber();
}
```
That way you can trust your non-strict comparisons, as well as perform relational abstract comparisons (e.g., `<=`), thus possibly avoiding a lot of `toNumber()` calls.